sdk/filter/filter.go

```diff
@@ -9,4 +9,8 @@ type Filter interface {
 
 	// EvaluateBody can be used to evaluate the body content
 	EvaluateBody(span sdk.Span, body []byte) bool
+
+	// Evaluate can be used to evaluate the span attributes. Notice span will contain at most
+	// pre-processing attributes (e.g. request attributes in a HTTP/RPC call).
+	Evaluate(span sdk.ReadbackSpan) bool
```

Comment:
This method allows to filter based on the span attributes. The reason why we keep this method along with the others is because the easiest cases are to evaluate by header, url and body whereas the span attributes requires the instrumentation to create a ReadbackSpan cc @tyroneyu @ryandens

Reply:
Regarding the existing two interface methods, we still have the problem with
https://traceableai.atlassian.net/wiki/spaces/Engineering/pages/1265139838/Blocking+rules+-+evaluation+order+of+precedence right?  Specifically when we allow an IP address since the required headers aren't being sent to either call.